morpheus/stages/general/nonlinear_modules_stage.py

```diff
@@ -0,0 +1,119 @@
+from morpheus.utils.module_utils import load_module
+
+logger = logging.getLogger(__name__)
+
+
+class NonLinearModulesStage(Stage):
+    """
+    Loads an existing, registered, MRC SegmentModule and wraps it as a Morpheus Stage.
+
+    Parameters
+    ----------
+    c : `morpheus.config.Config`
+        Pipeline configuration instance.
+    module_config : typing.Dict
+        Module configuration.
+    input_port_name : str
+        Name of the input port for the registered module.
+    output_port_name_prefix : str
+        Prefix name of the output ports for the registered module.
+    output_port_count : str
+        Number of output ports for the registered module.
+    input_type : default `typing.Any`
+        The stage acceptable input type.
+    output_type : default `typing.Any`
+        The output type that the stage produces.
+
+    """
+
+    def __init__(self,
+                 c: Config,
+                 module_config: typing.Dict,
+                 input_port_name: str,
+                 output_port_name_prefix: str,
+                 output_port_count: int,
+                 input_type=typing.Any,
+                 output_type=typing.Any):
+
+        super().__init__(c)
+
+        self._input_type = input_type
+        self._ouput_type = output_type
+        self._module_config = module_config
+        self._input_port_name = input_port_name
+        self._output_port_name_prefix = output_port_name_prefix
+
+        assert output_port_count > 0, "Output port count must be >= 1"
+
+        self._create_ports(1, output_port_count)
+        self._output_port_count = output_port_count
+
+    @property
+    def name(self) -> str:
+        return self._module_config.get("module_name", "non_linear_module")
+
+    def supports_cpp_node(self):
+        return False
+
+    def input_types(self) -> typing.Tuple:
+        """
+        Returns input type for the current stage.
+        """
+
+        return (typing.Any, )
+
+    def accepted_types(self) -> typing.Tuple:
+        """
+        Accepted input types for this stage are returned.
+
+        Returns
+        -------
+        typing.Tuple
+            Accepted input types.
+
+        """
+        return (typing.Any, )
+
+    def _build(self, builder: mrc.Builder, in_stream_pairs: typing.List[StreamPair]) -> typing.List[StreamPair]:
+
+        assert len(in_stream_pairs) == 1, "Only 1 input is supported"
+
+        in_stream_node = in_stream_pairs[0][0]
+
+        # Laod module from registry.
+        module = load_module(self._module_config, builder=builder)
+        mod_in_stream = module.input_port(self._input_port_name)
+
+        builder.make_edge(in_stream_node, mod_in_stream)
+
+        out_stream_pairs = []
+
+        count = 0
+        while (count < self._output_port_count):
+            out_port = f"{self._output_port_name_prefix}-{count}"
+            out_stream_pairs.append((module.output_port(out_port), self._ouput_type))
+            count += 1
```

Comment:
This seems cumbersome and enforces a lot of rules on the port naming scheme. A better implementation would be to accept 2 lists in the constructor: `input_port_names` and `output_port_names`. Then in the constructor you can do:

```python
self._create_ports(len(input_port_names), len(output_port_names))
```
Then in the build function, assign the correct `in_stream_pairs` to the correct input ports using `input_port_names`. And then return the `out_stream_pairs` just like you are doing now.

This would support any port naming scheme and would be very flexible.